src/middlewared/middlewared/plugins/filesystem_/acl_base.py

```diff
@@ -292,11 +292,21 @@ def setacl(self, job, data):
     @accepts(
         Str('path'),
         Bool('simplified', default=True),
+        Bool('resolve_ids', default=False),
     )
-    def getacl(self, path, simplified):
+    def getacl(self, path, simplified, resolve_ids):
         """
         Return ACL of a given path. This may return a POSIX1e ACL or a NFSv4 ACL. The acl type is indicated
-        by the `ACLType` key.
+        by the `acltype` key.
+
+        `simplified` NFSv4 - returns a shortened form of the ACL permset, POSIX1E - this is a no-op.
+
+        `resolve_ids` - adds additional `who` key to each ACL entry, that converts the numeric id to
+        a user name or group name. In the case of owner@ and group@ (NFSv4) or USER_OBJ and GROUP_OBJ
+        (POSIX1E), st_uid or st_gid will be converted from stat() return for file. In all other cases.
+        MASK (POSIX1E), OTHER (POSIX1E), everyone@ (NFSv4), key `who` will be included, but set to null.
+        In case of failure to resolve the id to a name, `who` will be set to null. This option should
+        only be used if resolving ids to names is required.
```

Comment:
Query: would it be useful to convert an unknown id to its string representation ? say uid 12345 is unable to be translated into its `username`, would it be useful to return the string "12345" as the `who`  instead of simply returning null ?

Reply:
I'm on the fence about that one honestly. In JSON output variant of nfs4xdr_getfacl, I do just that (return numeric ID as string). Our webui team though puts the name in red text in case of name resolution failure and I thought this might be a good way to indicate that there is an issue to them. 

I was also thinking of cases where id and name might be identical. uid = 12345 and user gives name "12345".

Reply:
Solid point ! I know we had to do something similar in Solaris, but I realize this can open up Pandora's box. Only making a suggestion for visibility purposes (ie. it may be more useful to see a stringified uid than a null), but I feel your pain. 😬